renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -5069,6 +5069,10 @@ void GatherPSInputDataForInitialValues(const DXBC::DXBCContainer *dxbc,
                   (sig.regChannelMask & 0x4 ? 1 : 0) + (sig.regChannelMask & 0x8 ? 1 : 0);
 
     rdcstr name = sig.semanticIdxName;
+    if(sig.semanticIndex != 0)
+    {
+      name += ToStr(sig.semanticIndex);
+    }
```

Comment:
This fix does not seem correct. As in the comments on the field `semanticIdxName` it already includes the index when necessary as compared to `semanticName`, so this will duplicate it turning `TEXCOORD1` into `TEXCOORD11`.

Perhaps the logic for that needs to be re-examined? If you're not sure you can open this as a bug report and not a pull request and I can investigate it.

Reply:
> This fix does not seem correct. As in the comments on the field `semanticIdxName` it already includes the index when necessary as compared to `semanticName`, so this will duplicate it turning `TEXCOORD1` into `TEXCOORD11`.
> 
> Perhaps the logic for that needs to be re-examined? If you're not sure you can open this as a bug report and not a pull request and I can investigate it.

Oh, thanks - I see where I'm wrong =)
I believe root reason for issues goes from this code: 

```
for(uint32_t i = 0; i < sign->numElems; i++)
{
  SigParameter &a = (*sig)[i];

  for(uint32_t j = 0; j < sign->numElems; j++)
  {
    SigParameter &b = (*sig)[j];
    if(i != j && a.semanticName == b.semanticName)
    {
      a.needSemanticIndex = true;
      break;
    }
  }

  rdcstr semanticIdxName = a.semanticName;
  if(a.needSemanticIndex)
    semanticIdxName += ToStr(a.semanticIndex);

  a.semanticIdxName = semanticIdxName;
  }
}
```
Which literally means, that we use semantic index as a part of semanticIdxName field only if semantic with given name is used more than once. However if we only use single semantic with given name, but non-zero index, this doesn't work =(

Reply:
That seems to make more sense, it sounds quite unusual to only use a higher numbered semantic but I can see how that would break here.

If you want to update the PR that is fine, please ensure that as noted above you follow the [CONTRIBUTING guidelines](https://github.com/baldurk/renderdoc/blob/v1.x/docs/CONTRIBUTING/Preparing-Commits.md) and remove the old commits and just have a new commit with the correct fix.